snakemake_executor_plugin_slurm/utils.py

```
@@ -14,3 +15,37 @@
+
+def delete_empty_dirs(path: Path) -> list:
+    """
+    Function to delete all empty directories in a given path.
+    This is needed to clean up the working directory after
+    a job has sucessfully finished. This function is needed because
+    the shutil.rmtree() function does not delete empty
+    directories.
+    """
+
+    # get a list of all directorys in path and subpaths
+    def get_dirs(path: Path, result=[]):
```

Ruff (0.8.2)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)